### ISSUE TITLE
[Test Failure] F5 gives error at runtime on MVC and Web Forms projects

### ISSUE THREAD

**Issue by v-nancyj** [NuGet/Home#4769]
**NuGet Version:** NuGet 4.1.0-rtm-2357 **(Insertable)**
**VS Version:** d15prstg1 26307.01
**OS Version:** Windows 10 RS2_Release 15051.0.170303-1951

**Repro Steps:**
1.	Click File -> New project -> Visual C# -> Web -> ASP.NET Web Application.
2.	Click MVC template.
3.	Select deploy target to IE and build the solution, no error.
4.	Press F5.

**Expected Result:**
No error occur.

**Actual Result:**
Fail to deploy MVC WebApplication on IE with the error as below screenshot.
![111](https://cloud.githubusercontent.com/assets/20632775/23694668/0eb07582-0416-11e7-88fc-e33d7b3a4355.png)
 

**Stack Trace:**
[FileLoadException: Could not load file or assembly 'Newtonsoft.Json' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]

[FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +0
   System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +36
   System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +152
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean forIntrospection) +77
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, Boolean forIntrospection) +21
   System.Reflection.Assembly.Load(String assemblyString) +28
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +38

[ConfigurationErrorsException: Could not load file or assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +738
   System.Web.Configuration.CompilationSection.LoadAllAssembliesFromAppDomainBinDirectory() +217
   System.Web.Configuration.CompilationSection.LoadAssembly(AssemblyInfo ai) +130
   System.Web.Compilation.BuildManager.GetReferencedAssemblies(CompilationSection compConfig) +170
   System.Web.Compilation.BuildManager.GetPreStartInitMethodsFromReferencedAssemblies() +92
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +290
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +157
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +549

[HttpException (0x80004005): Could not load file or assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +10044576
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +95
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

**Notes:**
1.	Then type in ‘Update-Package’ in PMC, hit F5, the MVC WebApplication can be deployed in IE successfully as below screenshot.
![222](https://cloud.githubusercontent.com/assets/20632775/23694693/39f08610-0416-11e7-9cd9-11bd5a31c8c6.png)

2.	The issue repro on Web Forms WebApplication.

3.	The issue repro on **Microsoft Edge**.

**Comment by jainaashish** [comment]
Cannot repro with VS 2017 RTW + NuGet 4.1.0-rtm-2357

Can you check if it's specific to d15prstg1 branch build? and also share machine details where this bug can be repro'd

**Comment by v-nancyj** [comment]
[Repro machine:EXPRBLNVM81177](http://maddog/goto/MDGoto.Application?md://OrcasTS/Machine/1204148) 
I have add your alias to the machine.

**Comment by jainaashish** [comment]
Can't access the machine. Also Did it repro with any other VS branch build?

**Comment by v-nancyj** [comment]
@jainaashish Repro machine: [EXPRBLNVM30551](http://maddog/goto/MDGoto.Application?md://OrcasTS/Machine/707025)
This VS build is 26309.01, and the NuGet is default version 4.1.0-rtm-2353, the issue also repro on this buid

**Comment by jainaashish** [comment]
@v-nancyj I asked if you were able to repro this issue except d15prstg1 branch? 26309.01 is also from d15prstg1 VS branch...

**Comment by jainaashish** [comment]
Ok I logged into the machine, and saw there is some issue with BindingRedirects which is why it's failing to get the right version of Newtonsoft.json 

Now, I tried to repro this issue at multiple machines with same VS/ NuGet versions but couldn't do so... which makes me believe that there is something wrong with the machine itself the way VS is being installed there. Somehow It's getting newtonsoft.json 8.0.4 version as latest from framework assemblies which is why it updates the existing binding redirects from 6.0.0 to 8.0.4 which causes this issue.

Install VS on a clean machine and then try to repro it.
